### PR TITLE
[UM-2]-create-course-class

### PR DESCRIPTION
**Pull request by HKalatas**
Course tablosu ticket ta belirtilen attribute lar ile oluşturuldu.  displayEnrolledStudents metodu yazıldı. ilgili getter-setter metodları oluşturuldu. 

Ekran görüntüsü 2024-05-18 171948
Ekran görüntüsü 2024-05-18 172314

### CODE REVIEW

File: src/main/java/com/archisacadeny/course/Course.java [hunk 1]
```diff
@@ -1,7 +1,53 @@
...
 import com.archisacadeny.config.DataBaseConfig;
+import com.archisacadeny.student.Student;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class Course {
+    private long id;
+    private String courseName;
+    private long instructorId;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/course/Course.java]:
```suggestion
    private Instructor instructor;
```

File: src/main/java/com/archisacadeny/instructor/Instructor.java [hunk 2]
```diff
@@ -1,4 +1,50 @@
 package com.archisacadeny.instructor;
 
 public class Instructor {
+
+    private long id;
+    private String fullName;
+    private String gender;
+    private int courseId;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/instructor/Instructor.java]: private List<Course>  courses;

File: src/main/java/com/archisacadeny/instructor/InstructorRepository.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class InstructorRepository {
+    public static void createInstructorTable(){
+        try(Statement statement = DataBaseConnectorConfig.getConnection().createStatement())
+        {
+            // SQL query for creating instructor table
+            String query = "CREATE TABLE IF NOT EXISTS instructors(" +
+                    "id INTEGER PRIMARY KEY," +
+                    "fullName VARCHAR(255)," +
+                    "gender VARCHAR(255), " +
+                    "courseId INTEGER," +
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/instructor/InstructorRepository.java]: we don't need courseId in this table

File: src/main/java/com/archisacadeny/student/Student.java [hunk 4]
```diff
@@ -1,4 +1,76 @@
 package com.archisacadeny.student;
 
 public class Student {
+    private long id;
+    private String fullName;
+    private String gender;
+    private long courseID;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/Student.java]:
```suggestion
    private List<Course> enrolledCourses;
```

File: src/main/java/com/archisacadeny/course/Course.java [hunk 5]
```diff
@@ -1,7 +1,53 @@
...
+
+    private ArrayList<Student> enrolledStudents;
+
+    public void displayEnrolledStudents() {
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/course/Course.java]: we don't need this method inside this class

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/archisacadeny/course/Course.java [hunk 6]
```diff
@@ -1,7 +1,38 @@
...
+import com.archisacadeny.student.Student;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class Course {
+    private long id;
+    private String courseName;
+    private Instructor instructor;
+
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/course/Course.java:33]: we should add other fields for this class such as:
credit
total student
enrolledStudents
department

File: src/main/java/com/archisacadeny/course/CourseRepository.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class CourseRepository {
+    public static void createCourseTable(){
+        try(Statement statement = DataBaseConnectorConfig.getConnection().createStatement())
+        {
+            // SQL query for creating courses table
+            String query = "CREATE TABLE IF NOT EXISTS courses(" +
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/course/CourseRepository.java]: update the query according to the fields

File: src/main/java/com/archisacadeny/instructor/Instructor.java [hunk 8]
```diff
@@ -1,4 +1,52 @@
...
 
+import java.awt.*;
+
 public class Instructor {
+
+    private long id;
+    private String fullName;
+    private String gender;
+    private String department;
+    private List courses;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/instructor/Instructor.java]:
```suggestion
    private List<Course> courses;
```

File: src/main/java/com/archisacadeny/instructor/Instructor.java [hunk 9]
```diff
@@ -1,4 +1,52 @@
 package com.archisacadeny.instructor;
 
+import java.awt.*;
+
 public class Instructor {
+
+    private long id;
+    private String fullName;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/instructor/Instructor.java]: private String indentityNo;

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/archisacadeny/instructor/Instructor.java (hunk 9 above)
Comment (HKalatas) [reply src/main/java/com/archisacadeny/instructor/Instructor.java]: CourseStudentMapper tablosu eklendi

File: src/main/java/com/archisacadeny/course/Course.java [hunk 10]
```diff
@@ -1,7 +1,93 @@
...
+import com.archisacadeny.instructor.Instructor;
+import com.archisacadeny.student.Student;
+import java.util.List;
+import java.util.ArrayList;
 
 public class Course {
+    private long id;
+    private String courseName;
+    private Instructor instructor;
+    private long creditHours;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/course/Course.java]:
```suggestion
    private long credits;
```

File: src/main/java/com/archisacadeny/course/Course.java [hunk 11]
```diff
@@ -1,7 +1,93 @@
...
+        this.id = id;
+        this.courseName = courseName;
+        this.instructor = instructor;
+        this.creditHours = creditHours;
+        this.courseNumber = courseNumber;
+        this.enrolledStudents = enrolledStudents;
+        this.department = department;
+        this.maxStudents = maxStudents;
+    }
+
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/course/Course.java:33]: ```public Course(){}```

File: src/main/java/com/archisacadeny/courseStudentMapper/CourseStudentMapper.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
+package com.archisacadeny.courseStudentMapper;
+
+import com.archisacadeny.instructor.Instructor;
+import com.archisacadeny.student.Student;
+
+public class CourseStudentMapper {
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/courseStudentMapper/CourseStudentMapper.java]: we don't need this class

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/archisacadeny/course/Course.java (hunk 6 above)
Comment (HKalatas) [reply src/main/java/com/archisacadeny/course/Course.java:33]: Already added

File: src/main/java/com/archisacadeny/Main.java [hunk 13]
```diff
@@ -1,6 +1,5 @@
 package com.archisacadeny;
 
-import com.archisacadeny.config.DataBaseConfig;
 import com.archisacadeny.config.DataBaseConnectorConfig;
 
 import com.archisacadeny.course.Course;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/Main.java:5]: please resolve the conflict in order to merge this

**Review by JamshidS — CHANGES_REQUESTED** [review]

**Pull request closed without merging**